app/grandchallenge/evaluation/templates/evaluation/evaluation_list.html

```diff
@@ -17,6 +17,25 @@
     </ol>
 {% endblock %}
 
+{% block topbar2 %}
+    <div class="nav-pill-dropdown-container">
+        <a class="d-lg-none btn btn-outline-dark dropdown-toggle phaseButton mr-1 mb-3 px-4"
+           data-toggle="dropdown"
+           href="#"
+           role="button"
+           aria-expanded="false">
+            Choose a Phase</a>
+        <ul class="nav nav-pills col-12 mb-3">
+            {% for phase in challenge.phase_set.all %}
+                <li class="nav-item">
+                    <a class="nav-link mr-1 px-4 py-1 {% if phase.slug in phase_list %}active{% endif %}"
+                       href="{% url 'evaluation:leaderboard' challenge_short_name=challenge.short_name slug=phase.slug %}">{{ phase.title }}</a>
+                </li>
+            {% endfor %}
+        </ul>
+    </div>
```

Comment:
I think the active setting can be removed as the active represents a link to the leaderboard, so should only be set to active if the view is of the leaderboard detail for that phase. This way, you can remove the duplication and the extra context data too.